test/end-to-end/test_pkg_bulkupload.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Basic set of tests for the hab pkg bulkupload command
+#
+# Assumptions:
+# 1. ACCEPTANCE_HAB_AUTH_TOKEN or HAB_AUTH_TOKEN Environment variables are set and valid
+# 2. ${CACHE_DIR} can be set to a writable location on the filesystem
+# 3. non zero exit code from each command implies success
+
+# TODO: Future improvement would be to create a clean room Builder or origin.
+# Doing so would allow for more accurate validation of the bulkupload by interrogating
+# the Builder. Of course this introduces significant setup time cost..
+
+set -euo pipefail
+
+export HAB_NOCOLORING=true
+export HAB_NONINTERACTIVE=true
+export HAB_AUTH_TOKEN=${ACCEPTANCE_HAB_AUTH_TOKEN:-${HAB_AUTH_TOKEN}}
+export HAB_ORIGIN="testbulkupload"
+export HAB_BLDR_URL=${HAB_BLDR_URL:-https://bldr.acceptance.habitat.sh}
+
+HAB=${HAB_TEST_CMD:-hab}
+CACHE_DIR="test-cache"
+FIXTURES_DIR="fixtures/bulkupload"
+PKG_A_TAR="${FIXTURES_DIR}/testbulkupload-testpkg1-0.1.0-20191024190939.tar"
+PKG_B_TAR="${FIXTURES_DIR}/testbulkupload-testpkg2-0.1.0-20191024191005.tar"
+PKG_A_HART="${CACHE_DIR}/artifacts/testbulkupload-testpkg1-0.1.0-20191024190939-x86_64-linux.hart"
+PKG_B_HART="${CACHE_DIR}/artifacts/testbulkupload-testpkg2-0.1.0-20191024191005-x86_64-linux.hart"
+
+echo
+echo "==========Testing with command ${HAB}, using cache dir ${CACHE_DIR}"
+echo
+
+before_upload() {
+    echo
+    echo ">>>>> before_upload() tasks"
+    rm -rf ${CACHE_DIR}
+    # origin create will exit 0 if the origin already exists
+    ${HAB} origin create ${HAB_ORIGIN}
+    ${HAB} origin key download --secret ${HAB_ORIGIN}
+    mkdir -p ${CACHE_DIR}/artifacts ${CACHE_DIR}/keys
+    cp -f ~/.hab/cache/keys/${HAB_ORIGIN}-*pub ${CACHE_DIR}/keys/
+    ${HAB} pkg sign --origin ${HAB_ORIGIN} ${PKG_A_TAR} ${PKG_A_HART}
+    ${HAB} pkg sign --origin ${HAB_ORIGIN} ${PKG_B_TAR} ${PKG_B_HART}
+    echo
+}
+
+test_expecting_fail() {
+    DESC=$1
+    CMD=$2
+
+    echo
+    echo "==========Expected failure: Testing ${DESC}"
```

Comment:
Not sure if it's appropriate on this particular line, but you can prefix your `echo` strings with "--- " to give it a nice fold in the buildkite UI.   I've found this to be useful to break things up logically so that you can quickly find any output if you need to inspect the build logs

Reply:
Applied your suggestion in https://github.com/habitat-sh/habitat/pull/7113/commits/4df23afc31faf15a0324e1542128dc6a2f1357ad